### DISCUSSION TITLE
global._mongoClientPromise: Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature

### DISCUSSION THREAD

**Discussion by myhendry** [nextauthjs/next-auth#3320] [Help]
### Question 💬

How do we type global._mongoClientPromise?

When i type it as any, (global as any)._mongoClientPromise, i get a database connection error unhandledRejection: TypeError: Cannot read property 'db' of undefined

When I leave the codes untouched global._mongoClientPromise, the db connection works fine. However, there are a number of red warning lines in my code with the following warning Element implicitly has an 'any' type because type 'typeof globalThis' has no index signature.

### How to reproduce ☕️

if (process.env.NODE_ENV === "development") {
  // In development mode, use a global variable so that the value
  // is preserved across module reloads caused by HMR (Hot Module Replacement).

  // todo global._mongoClientPromise
  if ((!global as any)._mongoClientPromise) {
    client = new MongoClient(uri, options);
    (global as any)._mongoClientPromise = client.connect();
  }
  //@ts-ignore
  clientPromise = (global as any)._mongoClientPromise;
} else {
  // In production mode, it's best to not use a global variable.
  client = new MongoClient(uri, options);
  clientPromise = client.connect();
}

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by victorfu** [comment]
check this discussion
https://github.com/prisma/prisma/discussions/10037#discussioncomment-1569084

and you can use a similar way like the following:

```
  let globalWithMongo = global as typeof globalThis & {
    _mongoClientPromise: Promise<MongoClient>
  }
  if (!globalWithMongo._mongoClientPromise) {
    client = new MongoClient(uri)
    globalWithMongo._mongoClientPromise = client.connect()
  }
  clientPromise = globalWithMongo._mongoClientPromise
  ```

  For me, it solves the problem :D

**Reply by wojnet** [reply]
thank you!

**Comment by blazejmojak** [comment]
Thank you too!!! Great work.

**Comment by TejasThombare20** [comment]
Thank you
